sdk/resource_locator.go

```diff
@@ -159,7 +160,10 @@ func (rl ResourceLocator) GetIdentifier() (string, error) {
 		if rl.identifier == "" {
 			return "", fmt.Errorf("no resource locator identifer: %d", rl.protocol)
 		}
-		return rl.identifier, nil
+		// remove padding
+		cleanedIdentifier := strings.TrimRight(rl.identifier, "\x00")
+		cleanedIdentifier = strings.TrimLeft(cleanedIdentifier, "\x00")
```

Comment:
We should probably decide if we want to pad left or right and just do one, but I'm fine with this for now.

Reply:
agreed.  rewrap is more robust (within reason and with security considerations) and the SDK should be more strict and uniform